cmd/minishift/cmd/start_preflight.go

```diff
@@ -0,0 +1,278 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/docker/machine/libmachine/drivers"
+	startFlags "github.com/minishift/minishift/cmd/minishift/cmd/config"
+	miniutil "github.com/minishift/minishift/pkg/minishift/util"
+
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/viper"
+)
+
+const (
+	StorageDisk = "/mnt/sda1"
+)
+
+type preflightCheckRunner func() bool
+type preflightCheckWithDriverRunner func(driver drivers.Driver) bool
+
+func preflightCheckSucceedsOrFails(configNameOverrideIfSkipped string, execute preflightCheckRunner, message string, treatAsWarning bool, configNameOverrideIfWarning string, errorMessage string) {
```

Comment:
I think it would be good for testing purposes as well as eventually maybe directing these kind of output to a log file to pass a Writer to these function. Per default it can be os.Stdout, but the nice thing is that it can also be something different. We are applying this approach already at some of the other new functionality (groups).